src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -247,20 +247,20 @@ unsigned HWIntrinsicInfo::lookupSimdSize(Compiler* comp, NamedIntrinsic id, CORI
 
     CORINFO_CLASS_HANDLE typeHnd = nullptr;
 
-    if (JITtype2varType(sig->retType) == TYP_STRUCT)
-    {
-        typeHnd = sig->retTypeSigClass;
-    }
-    else if (HWIntrinsicInfo::BaseTypeFromFirstArg(id))
+    if (HWIntrinsicInfo::BaseTypeFromFirstArg(id))
```

Comment:
Why is the change in order needed here?

Reply:
@echesakovMSFT explained that in his description:
> Before the change, if an intrinsic returns a struct type value - then the size of the value will be used for determining SIMD size. After the change, if an intrinsic has a flag BaseTypeFromFirstArg or BaseTypeFromSecondArg specified - then the SIMD size of the tree node will be based on the corresponding parameter size (i.e op1 or op2 size). For example, AddAcross always returns Vector64 value and the SIMD size cannot be based on the return value size (since it will be the same for AddAcross(Vector64<T>) and AddAcross(Vector128<T>))

Reply:
This is needed for `AddAcross`, `MaxAcross`, `AddPairwiseScalar` etc:

For example, `AddPairwiseScalar` returns `Vector64<T>` value and before that change the SIMD size computed here was based only on the return value size (even when `BaseTypeFromFirstArg` or `BaseTypeFromSecondArg` flags were set), i.e. it was 8 bytes in all of these cases:
* `Vector64<float> AddAcross(Vector128<float> value)` - SIMD size should be 16 bytes `FADDV Sd, Vn.4S`
* `Vector64<float> AddPairwiseScalar (Vector64<float> value)` - SIMD size is 8 bytes `FADDP Sd, Vn.2S`
* `Vector64<double> AddPairwiseScalar (Vector128<double> value)` - SIMD size should be 16 bytes  `FADDP Dd, Vn.2D`

Reply:
Sorry, I had completed missed that point.
Thanks for calling it out :)